app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocSyntaxHighlighter.java

```diff
@@ -0,0 +1,324 @@
+    private static final int AD_BACKCOLOR_LIGHT_SIDEBAR = TOL_PALE_RED;
+    private static final int AD_BACKCOLOR_DARK_SIDEBAR = TOL_DARK_RED;
+    private static final int AD_BACKCOLOR_LIGHT_TABLE = TOL_PALE_YELLOW;
+    private static final int AD_BACKCOLOR_DARK_TABLE = TOL_DARK_YELLOW;
+    private static final int AD_BACKCOLOR_LIGHT_ATTRIBUTE = TOL_PALE_CYAN;
+    private static final int AD_BACKCOLOR_DARK_ATTRIBUTE = TOL_DARK_CYAN;
+    // we use gray for miscellaneous to avoid too much variety
+    private static final int AD_BACKCOLOR_LIGHT_MONOSPACE = TOL_PALE_GRAY;
+    private static final int AD_BACKCOLOR_DARK_MONOSPACE = TOL_DARK_GRAY;
+    private static final int AD_BACKCOLOR_LIGHT_SQUAREBRACKETS = TOL_PALE_GRAY;
+    private static final int AD_BACKCOLOR_DARK_SQUAREBRACKETS = TOL_DARK_GRAY;
+    private static final int AD_BACKCOLOR_LIGHT_BLOCKTITLE = TOL_PALE_GRAY;
+    private static final int AD_BACKCOLOR_DARK_BLOCKTITLE = TOL_DARK_GRAY;
+
+    // use the same highlight for light and dark theme
+    private static final int AD_BACKCOLOR_LIGHT_HIGHLIGHT = Color.YELLOW;
+    private static final int AD_BACKCOLOR_DARK_HIGHLIGHT = Color.YELLOW;
+    // black on yellow
+    private static final int AD_FORECOLOR_HIGHLIGHT = Color.BLACK;
+
+    // ..._COLOR_LIGHT_... = Color for light theme
+    // TODO: too little contrast to normal font with the TOL_ colors
+//    private static final int AD_COLOR_LIGHT_COMMENT = TOL_DARK_GRAY;
+//    private static final int AD_COLOR_DARK_COMMENT = TOL_PALE_GRAY;
+//    // https://docs.oracle.com/javase/9/docs/api/javafx/scene/paint/Color.html
+//    // use Color.GRAY instead of TOL_GRAY, because Color.GRAY is the middle between white and black: #808080
+//    // alternative: TOL_PALE_GRAY vs. TOL_DARK_GRAY
+//    private static final int AD_COLOR_COMMENT = Color.GRAY; //TOL_GRAY;
+    private static final int AD_FORECOLOR_LIGHT_COMMENT = Color.GRAY;
+    private static final int AD_FORECOLOR_DARK_COMMENT = Color.GRAY;
+
+
+    private boolean _highlightLineEnding;
+    private boolean _highlightCodeChangeFont;
+    private boolean _highlightBiggerHeadings;
+    private boolean _highlightCodeBlock;
+
+
+    public AsciidocSyntaxHighlighter(AppSettings as) {
+        super(as);
+    }
+
+    @Override
+    public SyntaxHighlighterBase configure(Paint paint) {
+        _highlightLineEnding = _appSettings.isAsciidocHighlightLineEnding();
+// TODO: does not work yet
+        _highlightBiggerHeadings = _appSettings.isAsciidocBiggerHeadings();
+        _highlightCodeChangeFont = _appSettings.isHighlightCodeMonospaceFont();
+        _highlightCodeBlock = _appSettings.isHighlightCodeBlock();
+        _delay = _appSettings.getAsciidocHighlightingDelay();
+        return super.configure(paint);
+    }
+
+    @Override
+    protected void generateSpans() {
+        createTabSpans(_tabSize);
+        // TODO: understand: what happens here?
+        createUnderlineHexColorsSpans();
```

Comment:
It underlines HexColors, just like the function is named after. `#FF000` will underline FF0000 in red.  Comment can be removed